teps/0008-trigger-crd.md

```diff
@@ -0,0 +1,434 @@
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+<!--
+What are the risks of this proposal and how do we mitigate. Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the WGs or subproject.
+-->
+1. Security issue - Wide permission to EventListener. In this scenario EventListener
+will function in the same way as controller. It will be the responsibility of Operator
+or admin to manage that.
+2. Log: How will user access the EventListener log to debug their events. This can be addressed via
+emitting kubernetes events by EventListener and tkn-cli.
+3. Resource Hog issue: A particular namespace or project could hog most of the Resources. Operator
+can handle this issue. This is similar to triggers controller or webhook. Operator can parition heavy user to its own EL. Also, ELs are stateless and should be horizontally scalable.
+
+
+### User Experience (optional)
+
+<!--
+Consideration about the user experience. Depending on the area of change,
+users may be task and pipeline editors, they may trigger task and pipeline
+runs or they may be responsible for monitoring the execution of runs,
+via CLI, dashboard or a monitoring system.
+
+Consider including folks that also work on CLI and dashboard.
+-->
+
+### Performance (optional)
+Path base EventListener will improve the performance of EventListener by directly targetting
+the trigger definition we want to target instead of needlessly processing trigger that we don't.
+
+At the same time, one poorly behaved customer/namespace could dominate an eventlistener and effect
+other namespaces, which wouldn't be the case if each namespace had their own eventlistener.
+Operator/Admin can handle this issue by horizontally scaling EventListener or partitioning EventListener.
+<!--
+Consideration about performance.
+What impact does this change have on the start-up time and execution time
+of task and pipeline runs? What impact does it have on the resource footprint
+of Tekton controllers as well as task and pipeline runs?
+
+Consider which use cases are impacted by this change and what are their
+performance requirements.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+In first phases, this involve introduction of TriggerCRD and  modification of
+EventListenerCRD to introduce selectors which searches for these Trigger resource.
+
```

Comment:
My understanding is:
1. Introduce Trigger CRD and let EL's use them via Reference
2. Add selectors (maybe only namespace names first)
3. Path based urls for triggers to Listeners
3. Add other selectors? labels

Reply:
Yes, done that now.